### PR TITLE
Add documentation for Google Dataproc

### PR DESCRIPTION
**Pull request by madanadit**

### CODE REVIEW

File: docs/en/compute/Google-Dataproc.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+customizations.
+Aside from the added performance benefits of caching, Alluxio also enables users to run compute 
+workloads against on-premise storage or even a different cloud provider's storage i.e. AWS, Azure
+Blob Store.
+
+## Prerequisites
+
+* Account with Cloud Dataproc API enabled
+* A GCS Bucket
+* gcloud CLI: Make sure that the AWS CLI is set up and ready with the required AWS Access/Secret key
```
Comment (ns1123) [inline docs/en/compute/Google-Dataproc.md]: Replace the AWS stuff here

File: docs/en/compute/Google-Dataproc.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+An GCS bucket can be configured as Alluxio's Root Under File System and to serve as the location for the
```
Comment (ns1123) [inline docs/en/compute/Google-Dataproc.md]: "A GCS bucket" vs "An GCS Bucket". Also replace "can be configured" with "is required"

File: docs/en/compute/Google-Dataproc.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+bootstrap script.
+If required, the root UFS can be reconfigured to be HDFS or any other supported under store.
+
+## Basic Setup
+
+When creating a Dataproc cluster, Alluxio can be installed using an
+[initialization action](https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/init-actions)
+
+1. The Alluxio initialization action is hosted in a publicly readable
+GCS location **gs://alluxio-public/alluxio.sh**.
```
Comment (ns1123) [inline docs/en/compute/Google-Dataproc.md]: We should standardize the format as "gs://alluxio-public/dataproc/2.0.1/alluxio-dataproc.sh" @Xenorith please weigh in on this.

File: docs/en/compute/Google-Dataproc.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+* A required argument when is the root UFS URI using **alluxio_root_ufs_uri**.
```
Comment (ns1123) [inline docs/en/compute/Google-Dataproc.md]: remove "when"

File: docs/en/compute/Google-Dataproc.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+It manages the deployment of various Hadoop Services and allows for hooks into these services for
+customizations.
+Aside from the added performance benefits of caching, Alluxio also enables users to run compute 
+workloads against on-premise storage or even a different cloud provider's storage i.e. AWS, Azure
+Blob Store.
+
+## Prerequisites
+
+* Account with Cloud Dataproc API enabled
+* A GCS Bucket
```
Comment (ns1123) [inline docs/en/compute/Google-Dataproc.md]: I think gcloud has some version of IAM so we don't need to specify credentials. We should mention that that service must be enabled for the project/dataproc instances.

File: docs/en/compute/Google-Dataproc.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+> Note: The default Alluxio Worker memory is set to 1/3 of the physical memory on the instance.
+If a specific value is desired, set `alluxio.worker.memory.size` in the provided
+`alluxio-site.properties` or in the additional options argument.
+
+## Run a Spark Job
+
+The Alluxio bootstrap also takes care of setting up Spark for you.
+To run a Spark application accessing data from Alluxio, simply refer to the path as
+`alluxio://<cluster_name>-m:19998/<path_to_file>`; where `<cluster_name>-m` is the dataproc master
```
Comment (ns1123) [inline docs/en/compute/Google-Dataproc.md]: We shouldn't specify the master host/port and should just use "alluxio:///". Master hostname/port information can be retrieved from `alluxio-site.properties`.

File: docs/en/compute/Google-Dataproc.md (hunk 5 above)
Comment (madanadit) [reply docs/en/compute/Google-Dataproc.md]: Our legacy gs connector does not read credentials from the credential store

File: docs/en/compute/Google-Dataproc.md (hunk 6 above)
Comment (madanadit) [reply docs/en/compute/Google-Dataproc.md]: sgtm

**Comment by madanadit** [comment]
Thanks @ns1123. PTAL

File: docs/en/compute/Google-Dataproc.md (hunk 5 above)
Comment (ns1123) [reply docs/en/compute/Google-Dataproc.md]: I see. Then we should mention that interoperability keys are a prerequisite for Alluxio on GCS until we decide if the new GCS UFS will be in OS or not.

File: docs/en/compute/Google-Dataproc.md [hunk 7]
```diff
@@ -24,9 +24,9 @@ Blob Store.
 
 * Account with Cloud Dataproc API enabled
 * A GCS Bucket
-* gcloud CLI: Make sure that the AWS CLI is set up and ready with the required AWS Access/Secret key
+* gcloud CLI: Make sure that the CLI is set up with any necessary credentials
```
Comment (ns1123) [inline docs/en/compute/Google-Dataproc.md]: Mention interoperability keys as a pre-req.

File: docs/en/compute/Google-Dataproc.md (hunk 3 above)
Comment (Xenorith) [reply docs/en/compute/Google-Dataproc.md]: 👍 though said bucket has yet to be created

Comment (Xenorith) [reply docs/en/compute/Google-Dataproc.md]: i lied. it already exists and has a script in the exact location as described...

File: docs/en/compute/Google-Dataproc.md (hunk 5 above)
Comment (ns1123) [reply docs/en/compute/Google-Dataproc.md]: sounds good

**Comment by madanadit** [comment]
alluxio-bot, merge this please.

**Pull request merged**